frontend/public/components/namespace.jsx

```diff
@@ -137,14 +137,36 @@ const ProjectTableHeader = () => {
 };
 ProjectTableHeader.displayName = 'ProjectTableHeader';
 
-const ProjectTableRow = ({obj: project, index, key, style}) => {
+const ProjectLink = connect(
+  null,
+  { setActiveNamespace: UIActions.setActiveNamespace },
+)(({ project, setActiveNamespace }) => (
+  <span className="co-resource-item co-resource-item--truncate">
+    <ResourceIcon kind="Project" />
+    <button
+      type="button"
+      className="co-resource-item__resource-name btn btn-link"
+      onClick={() => setActiveNamespace(project.metadata.name)}
+    >
+      {project.metadata.name}
+    </button>
+  </span>
+));
+const projectHeaderWithoutActions = () => _.dropRight(ProjectTableHeader());
+
+const ProjectTableRow = ({obj: project, index, key, style, customData = {}}) => {
   const requester = getRequester(project);
+  const { ProjectLinkComponent, actionsEnabled = true } = customData;
   return (
     <TableRow id={project.metadata.uid} index={index} trKey={key} style={style}>
       <TableData className={projectColumnClasses[0]}>
-        <span className="co-resource-item">
-          <ResourceLink kind="Project" name={project.metadata.name} title={project.metadata.uid} />
-        </span>
+        {customData && ProjectLinkComponent ? (
```

Comment:
In order to reuse the ProjectTableRow component in the dev console pages, I have used the customData prop to pass the custom Component to the first cell in the row. @spadgett what do you think about this approach?
cc: @christianvogt

Reply:
There are couple of ways this could be done. Not sure if one way is really better than the other considering this is sort of an internal implementation detail.